UnityProject/Assets/Scripts/Systems/Explosions/NodeTypes/ExplosionNode.cs

```diff
@@ -198,18 +230,24 @@ public virtual void DoInternalDamage(float strength, BodyPart bodyPart)
 			}
 		}
 
-		public void TimedEffect(Vector3Int position, float time, string effectName, OverlayType effectOverlayType, TileChangeManager tileChangeManager)
+		public UniTask TimedEffect(Vector3Int position, float time, string effectName, OverlayType effectOverlayType, TileChangeManager tileChangeManager)
 		{
 			//Dont add effect if it is already there
-			if (tileChangeManager.MetaTileMap.HasOverlay(position, TileType.Effects, effectName)) return;
+			if (tileChangeManager.MetaTileMap.HasOverlay(position, TileType.Effects, effectName)) return UniTask.CompletedTask;
 			tileChangeManager.MetaTileMap.AddOverlay(position, TileType.Effects, effectName);
 			var Position = position.ToWorld(tileChangeManager.MetaTileMap.matrix);
-			var fireLightSpawn = Spawn.ServerPrefab(tileChangeManager.MetaTileMap.matrix.ReactionManager.FireLightPrefab,Position );
-
-			fireLightSpawn.GameObject.GetComponent<UniversalObjectPhysics>().AppearAtWorldPositionServer(Position);
-			fireLightSpawn.GameObject.GetComponent< ScaleSync>().SetScale(Vector3.one * 30);
+			//TODO: Pool this because it's ruining performance when multiple explosions occur.
+			var fireLightSpawn = Spawn.ServerPrefab(tileChangeManager.MetaTileMap.matrix.ReactionManager.FireLightPrefab, Position);
+			var physics =
+				ComponentsTracker<UniversalObjectPhysics>.GetComponentFromGameObject(fireLightSpawn.GameObject);
```

Comment:
this will have zero performance change since the get components are not cashed yet, since it is a new prefab

Reply:
There are no GetComponents and testing said otherwise. We're down a whole 1MB in GC when 10 explosions from Gibtonites are triggered and the stuttering is shorter now.

`GetComponentFromGameObject` looks up a Dictionary for components that auto register themselves to it. Thus bypassing another unnecessary GetComponent call that already happen a lot on `Awake`.